lib/include/ert/nexus/nexus_plot.hpp

```diff
@@ -41,17 +42,38 @@ struct unexpected_eof : public std::runtime_error {
 
 class NexusPlot {
 public:
+    struct NexusClassItem {
+        std::string name;
+        int timestep;
+        float time;
+        int num_items;
+        int max_items;
+        int max_perfs;
+        std::map< std::string, float > var;
+
+        float operator[] (std::string s) {
+            return this->var[s];
+        }
+    };
+
+    typedef std::map< std::string, std::vector<NexusClassItem> > timestep_data;
+
     NexusPlot( const std::string& );
     NexusPlot( std::istream& );
 
     int32_t num_classes = 0;
     int32_t day, month, year;
     int32_t nx, ny, nz;
     int32_t ncomp;
-    std::vector<std::string> class_names;
-    std::vector<int32_t> vars_in_class;
-    std::vector< std::vector<std::string> > var_names;
+    std::vector< std::string > class_names;
+    std::map< std::string, int32_t > vars_in_class;
+    std::map< std::string, std::vector<std::string> > var_names;
 
+private:
+    std::map< int, timestep_data > plot_data;
+
+public:
+    std::map< int, timestep_data >& data();
```

Comment:
I would provide both the const and non-const version of this method. That would allow you to create const refs to an instance of this class to pass around to functions